src/Features/Core/Portable/Completion/CompletionServiceWithProviders.cs

```diff
@@ -283,6 +291,33 @@ private CompletionProvider GetProviderByName(string providerName)
                 (expandItemsAvailableFromTriggeredProviders || expandItemsAvailableFromAugmentingProviders));
         }
 
+        private bool ValidatePossibleTriggerCharacterSet(IEnumerable<CompletionProvider> triggeredProviders, Document document, SourceText text, int caretPosition)
+        {
+            var syntaxFactsService = document.Project.LanguageServices.GetService<ISyntaxFactsService>();
+            if (caretPosition > 0 && syntaxFactsService != null)
+            {
+                // The trigger character has already been inserted before the current caret position.
```

Comment:
Is this still true in terms of deletion? i.e. after the first typed char is deleted, we should still see completion list in this case, right?

Reply:
Will resolve by only running during the insertion case.